CMakeLists.txt

```diff
@@ -69,23 +68,31 @@ endif()
 set(LLVM_LIT_ARGS "${LIT_ARGS_DEFAULT}" CACHE STRING "Default options for lit")
 
 # CIRCT project.
-set(CIRCT_MAIN_SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/include ) # --src-root
-set(CIRCT_INCLUDE_DIR ${CMAKE_CURRENT_BINARY_DIR}/include ) # --includedir
+set(CIRCT_MAIN_SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR} ) # --src-root
+set(CIRCT_MAIN_INCLUDE_DIR ${CIRCT_MAIN_SRC_DIR}/include)
 
 set(CIRCT_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
-set(CIRCT_BINARY_DIR ${CMAKE_BINARY_DIR}/bin)
+set(CIRCT_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR})
+set(CIRCT_INCLUDE_DIR ${CMAKE_CURRENT_BINARY_DIR}/include )
```

Comment:
Are these changes relating to include paths separable from the AddCIRCT changes?

Reply:
The most important bit is changing what `CIRCT_BINARY_DIR` is set to.  This is needed to have the correct include directories in `CIRCTConfig.cmake`.  Without that change, the include directory is set to `build/bin/include` instead of `build/include`.

The line:
```cmake
set(CIRCT_CONFIG_INCLUDE_DIRS
  "${CIRCT_SOURCE_DIR}/include"
  "${CIRCT_BINARY_DIR}/include"
  )
```

leads to:
```cmake
set(CIRCT_INCLUDE_DIRS "/Users/andrewy/wsp/circt/include;/Users/andrewy/wsp/circt/build/bin/include")
```

This can probably be separated out into another change, as no one is relying on any of this anyways. Should I go ahead an do this?

Reply:
I think it would make things easier to understand.  Also a good thing to capture in the commit coments.

Reply:
These have been split in to another pr